lab_2_spellcheck/main.py

```diff
@@ -50,7 +87,35 @@ def calculate_jaccard_distance(token: str, candidate: str) -> float | None:
     In case of corrupt input arguments, None is returned.
     In case of both strings being empty, 0.0 is returned.
     """
+    if not isinstance(token, str):
+        return None
 
+    if not isinstance(candidate, str):
+        return None
+    
+    if not token and not candidate:
+        return 0.0
+
+    tokens_unique=set(token)
+    candidate_unique=set(candidate)
+
+    intersection=set()
+    for element in tokens_unique:
```

Comment:
check how to compare sets, https://pythonworld.ru/tipy-dannyx-v-python/mnozhestva-set-i-frozenset.html